docs/DeveloperGuide.md

```diff
@@ -1,15 +1,247 @@
+prepareAddModuleCommand(storage, userCommandSecondKeyword, puModules, universities),
+DeleteModuleCommand(storage, indexToRemove, modules), and HelpCommand(). The parser class will handle error checking by
+throwing InvalidCommandException if the user's input command does not match the specified format.
+
+The following class diagrams illustrates the relationship between the Parser class and the Command classes.
+- (TODO: finish up the rest of the command cases)
+![ParserSequenceDiagram.png](diagrams%2FParserSequenceDiagram.png)
+
+### Help Command
+
+The help command provides a list of commands and the commands' respective input format for the user.  
+> Syntax: /help
+
+The following sequence diagram shows the relationship between the classes involved when the delete command is called.
+
+![HelpCommandSequenceDiagram.png](diagrams%2FHelpCommandSequenceDiagram.png)
+
+### Delete Command
+
+The delete command removes a module from the user's saved list of modules that is specified by the user.  
+> Syntax: delete [_uniAbbreviation_]/[_index_]
+
+The following sequence diagram shows the relationship between the classes involved when the delete command is called.
+
+![DeleteModuleCommandSequenceDiagram.png](diagrams%2FDeleteModuleCommandSequenceDiagram.png)
+
+### List Current Command
+
+Lists out all of current modules that user has added.
+
+
+> Syntax: list current
+
+Sequence Diagram of List Current Command.
+
+![ListCurrentCommandSequenceDiagram.png](diagrams%2FListCurrentCommandSequenceDiagram.png)
+
+**Explanation**
+
+1. ListCurrentCommand calls printCurrentPuModList() of UI class, passing an ArrayList<Module> modules.
+2. UI object calls ListCurrentPu Command for all universities stored in the class-level object _universities_.
+3. UI prints out to userConsole the modules user has added to his list in order of the Partner Universities.
+
+See ListCurrentPuCommand for further explanation: 
+[ListCurrentPuCommand](#list-current-pu-command) 
+
+
+### List Pu Command
+
+Lists out all of Partner Universities.
+
+> Syntax: list pu
+
+Sequence Diagram of List Pu Command.
+
+
+![ListPuCommandSequenceDiagram.png](diagrams%2FListPuCommandSequenceDiagram.png)
+
+**Explanation**
+
+1. ListPuCommand calls printPUListMessage() of UI class, which simply prints out a string of message to user
+2. ListPuCommand calls printPUList() of UI class.
+3. UI class holds an instance of all the possible PUs in an Arraylist<University> object.
+4. UI class loops through ArrayList<University> to receive the various university information and print out
+   to UserConsole
+
+### List Pu Modules Command
+
+Prints out list of modules available at a given Partner University
+
+> Syntax: list [_uniAbbreviation_]
+
+**Note: Partner Universities Abbreviations can be found using List Pu command**
+
+Sequence Diagram of List Pu Modules Command.
+![ListPuModulesCommandSequenceDiagram.png](diagrams%2FListPuModulesCommandSequenceDiagram.png)
```

Comment:
I feel like your UML diagram is a bit big and filled with a lot of words. Maybe you could try to make it more concise by making it small and precise.